src/check_redis_differences.py

```diff
@@ -0,0 +1,227 @@
+# running Redis instance. It compares the configurations based on key-value
+# pairs and reports any discrepancies found.
+#
+# Copyright (c) 2024 InnoGames GmbH
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+import argparse
+import re
+import sys
+from typing import Union
+
+import redis
+
+# Nagios exit codes
+OK = 0
+WARNING = 1
+CRITICAL = 2
+UNKNOWN = 3
+
+
+def parse_redis_config(file_path: str) -> dict[str, Union[str, list[str]]]:
+    """
+    Parses a Redis configuration file and returns a dictionary of key-value pairs.
+    """
+    config = {}
+    # Some config values are multi-line, needs special treatment
+    multi_line_keys = ['save', 'client-output-buffer-limit']
+    current_key = None
+
+    try:
+        with open(file_path) as f:
+            for line in f:
+                line = line.strip()
+                if line and not line.startswith('#'):
+                    parts = line.split(maxsplit=1)
+                    if len(parts) == 2:
+                        key, value = parts
+                        if key in multi_line_keys:
+                            if key not in config:
+                                config[key] = []
+                            config[key].append(value)
+                        else:
+                            config[key] = value.strip('"')
+                    elif current_key in multi_line_keys:
+                        config[current_key].append(line)
+                    current_key = parts[0] if parts else None
+        return config
+    except Exception as e:
+        # Maybe permission error? But let's catch more broadly
+        print(f'UNKNOWN - Error: {str(e)}')
+        sys.exit(UNKNOWN)
+
+
+def normalize_value(value: Union[str, list[str]]) -> str:
+    """
+    Joins list values into a single string and converts to lowercase.
+    """
+    if isinstance(value, list):
+        return ' '.join(value)
+    return str(value).lower()
```

Comment:
Given your type hints you already expect only strings so the type cast is redundant.
```suggestion
    return value.lower()
```